browser/brave_wallet/solana_provider_browsertest.cc

```diff
@@ -0,0 +1,211 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/command_line.h"
+#include "base/feature_list.h"
+#include "base/memory/raw_ptr.h"
+#include "base/path_service.h"
+#include "base/test/bind.h"
+#include "base/test/scoped_feature_list.h"
+#include "brave/browser/brave_wallet/brave_wallet_tab_helper.h"
+#include "brave/browser/brave_wallet/keyring_service_factory.h"
+#include "brave/common/brave_paths.h"
+#include "brave/components/brave_wallet/browser/keyring_service.h"
+#include "brave/components/brave_wallet/common/features.h"
+#include "brave/components/permissions/contexts/brave_wallet_permission_context.h"
+#include "chrome/browser/content_settings/host_content_settings_map_factory.h"
+#include "chrome/browser/profiles/profile.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/content_settings/core/browser/host_content_settings_map.h"
+#include "components/network_session_configurator/common/network_switches.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "content/public/test/test_utils.h"
+#include "net/dns/mock_host_resolver.h"
+#include "net/test/embedded_test_server/embedded_test_server.h"
+#include "url/gurl.h"
+
+namespace brave_wallet {
+
+namespace {
+constexpr char first_account[] = "8J7fu34oNJSKXcauNQMXRdKAHY7zQ7rEaQng8xtQNpSu";
+}
+
+class SolanaProviderTest : public InProcessBrowserTest {
+ public:
+  SolanaProviderTest() : https_server_(net::EmbeddedTestServer::TYPE_HTTPS) {
+    scoped_feature_list_.InitAndEnableFeature(
+        brave_wallet::features::kBraveWalletSolanaFeature);
+  }
+
+  ~SolanaProviderTest() override = default;
+
+  void SetUpCommandLine(base::CommandLine* command_line) override {
+    InProcessBrowserTest::SetUpCommandLine(command_line);
+    command_line->AppendSwitch(switches::kIgnoreCertificateErrors);
```

Comment:
Sorry I missed this in the last round, we should stop using this in new tests.
Please use `https_server_.SetSSLConfig(net::EmbeddedTestServer::CERT_TEST_NAMES);` and do s/a.com/a.test.
See https://bravesoftware.slack.com/archives/C7VLGSR55/p1636702893335800

Reply:
browser/permissions/permission_manager_browsertest.cc is an example

Reply:
fixed and squashed in fc0a3072f287883a6127e8423dd2182a5bd01e7a